pkgs/tools/filesystems/irods/common.nix

```diff
@@ -0,0 +1,74 @@
+{ stdenv, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive, avro-cpp, boost, jansson, zeromq, openssl, pam, libiodbc, kerberos, gcc, libcxx, cross, which }:
+
+# Common attributes of irods packages
+
+with stdenv;
+
+# Boost and avro-c++ have to be built with the same stdenv
```

Comment:
I think that I found the solution! I completely removed this part and simply added a boost override in the callPackages:

```
 inherit (callPackages ../tools/filesystems/irods rec {   
           stdenv = llvmPackages_38.libcxxStdenv;         
           libcxx = llvmPackages_38.libcxx;               
           boost = boost160.override { inherit stdenv; }; 
         })                                               
     irods                                                
     irods-icommands;                                     
```

I re-tested my binaries. It works!